runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerPlugin.java

```diff
@@ -0,0 +1,249 @@
+	private static final Dimension DEFAULT_SIZE = new Dimension(2, 2);
+
+	@Inject
+	private EventBus eventBus;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Inject
+	private MouseManager mouseManager;
+
+	@Inject
+	private PluginToolbar pluginToolbar;
+
+	@Inject
+	private ScreenMarkerCreationOverlay overlay;
+
+	@Inject
+	private OverlayRenderer overlayRenderer;
+
+	private ScreenMarkerMouseListener mouseListener;
+	private ScreenMarkerPluginPanel pluginPanel;
+	private NavigationButton navigationButton;
+
+	@Getter
+	private List<ScreenMarker> screenMarkers = new ArrayList<>();
+
+	@Getter(AccessLevel.PACKAGE)
+	private ScreenMarker currentMarker;
+
+	@Getter
+	private boolean creatingScreenMarker = false;
+
+	@Override
+	public Collection<Overlay> getOverlays()
+	{
+		List<Overlay> overlays = new ArrayList<>();
+		overlays.add(overlay);
+
+		for (ScreenMarker marker : screenMarkers)
+		{
+			ScreenMarkerOverlay screenMarkerOverlay = new ScreenMarkerOverlay(marker);
+			overlays.add(screenMarkerOverlay);
+		}
+
+		return overlays;
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		screenMarkers = loadConfig(null);
+		pluginPanel = injector.getInstance(ScreenMarkerPluginPanel.class);
+		pluginPanel.init();
+
+		BufferedImage icon;
+		synchronized (ImageIO.class)
+		{
+			icon = ImageIO.read(ScreenMarkerPlugin.class.getResourceAsStream(ICON_FILE));
+		}
+
+		navigationButton = NavigationButton.builder()
+			.tooltip(PLUGIN_NAME)
+			.icon(icon)
+			.panel(pluginPanel)
+			.build();
+
+		pluginToolbar.addNavigation(navigationButton);
+
+		mouseListener = new ScreenMarkerMouseListener(this);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		pluginToolbar.removeNavigation(navigationButton);
+		setMouseListenerEnabled(false);
+		creatingScreenMarker = false;
+		screenMarkers.clear();
+
+		pluginPanel = null;
+		currentMarker = null;
+		mouseListener = null;
+		navigationButton = null;
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (screenMarkers.isEmpty() && event.getGroup().equals(CONFIG_GROUP) && event.getKey().equals(CONFIG_KEY))
+		{
+			screenMarkers = loadConfig(event.getNewValue());
+		}
+	}
+
+	public void setMouseListenerEnabled(boolean enabled)
+	{
+		if (enabled)
+		{
+			mouseManager.registerMouseListener(mouseListener);
+		}
+		else
+		{
+			mouseManager.unregisterMouseListener(mouseListener);
+		}
+	}
+
+	public void startCreation(Point location)
+	{
+		currentMarker = new ScreenMarker(
+			new Rectangle(location, DEFAULT_SIZE),
+			DEFAULT_MARKER_NAME,
+			pluginPanel.getSelectedBorderThickness(),
+			pluginPanel.getSelectedColor(),
+			pluginPanel.getSelectedFillColor(),
+			true
+		);
+		creatingScreenMarker = true;
+	}
+
+	public void finishCreation(boolean aborted)
+	{
+		if (!aborted)
+		{
+			setMouseListenerEnabled(false);
+			pluginPanel.setCreationEnabled(false);
+			screenMarkers.add(currentMarker);
```

Comment:
Isn't this attempting to add to a fixed-size list? Unless I'm missing something.